src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/NewAzureStorageAccount.cs

```diff
@@ -176,7 +176,10 @@ public override void ExecuteCmdlet()
             if (this.EnableEncryptionService != null)
             {
                 createParameters.Encryption = ParseEncryption(EnableEncryptionService);
-                createParameters.Encryption.KeySource = "Microsoft.Storage";
+                if (createParameters.Encryption != null)
```

Comment:
Covering test?  Also, rather than checking this way, you can add a ValidateNotNullOrEmpty attribute to this parameter and here check for ```MyInvocation.BoundParameters.ContainsKey('Encryption')``` which is the recommended pattern if you want to see if a parameter was bouynd from either user input or the pipeline

Reply:
Hi Mark,
As you know, generally we will not cover the data plan cmldets in PowerShell unit test framework since XSCL request can't be record in unit test. I have added cases to test Encryption as none in our team regression test. (let me know if you would like to see the CR for the test).

ValidateNotNullOrEmpty or MyInvocation.BoundParameters.ContainsKey() not work for this issue, since when set "-EnableEncryptionService None" it can pass both check, so will still get the issue.

We will make the "-EnableEncryptionService" as obsolete very soon as feature team requested. I will send a PR for this soon after get the detail requirement from feature team. But before that, would like to fix this issue first.

Reply:
@blueww The problem there is that people wanting to contribute to these cmdlets sdo not have access to these tests, and so cann't tell if there is a regression.  Please add a covering test here for this issue

Reply:
@markcowl  
Unit test is added. please check.